hazelcast-client/src/test/java/com/hazelcast/client/map/ClientMapPartitionLostListenerTest.java

```diff
@@ -88,7 +89,10 @@ public void test_mapPartitionLostListener_invoked() {
         warmUpPartitions(instance, client);
 
         final EventCollectingMapPartitionLostListener listener = new EventCollectingMapPartitionLostListener(0);
-        client.getMap(mapName).addPartitionLostListener(listener);
+        // for a map partition lost event to be published, the map container must have been created
+        IMap map = client.getMap(mapName);
+        map.put("a", "b");
+        map.addPartitionLostListener(listener);
```

Comment:
Imagine a cluster mix of lite and data-members if all data members are lost, a lite-member master can not inform cluster for lost partitions. Because it doesn't know any map-container.

According to this example scenario, it seems to me, adding a `partitionLostListener` should be enough to get partition-lost events, an additional put to create a map-container is an unexpected thing + a behavioural change compared to existing 3.6 releases.